pkgs/development/python-modules/geomag/default.nix

```
@@ -0,0 +1,35 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pythonOlder
+, functools32
+, pytestCheckHook
+}:
+
+buildPythonPackage rec {
+  pname = "geomag";
+  version = "2020-05-27";
+
+  src = fetchFromGitHub {
+    owner = "rinigus";
+    repo = "geomag";
+    rev = "8eb9a730c8643fb7d63fdee4fd9a195ee8ba4df2";
+    sha256 = "1c1s75g3xfjib396y486sdbqsp4rkpvvcla3j2221symdx1l29s2";
+  };
+
+  propagatedBuildInputs = lib.optional (pythonOlder "3.2") functools32;
```

```suggestion
  propagatedBuildInputs = lib.optionals (pythonOlder "3.2") [ functools32 ];
```

FYI for future PRs: this is preferred because it is more strict and less likely to do mistakes with.

---

What's more strict about it? I don't see any advantage.

---

optionals makes sure that the 2nd argument is a list. optional could be anything.

---

Pedantically speaking, they both return a list (either empty or containing the second argument), and I don't think either cares what the types of the list contents are, so there is no meaningful difference in terms of types: `lib.optional true garbage` returns `[ garbage ]`, while `lib.optionals true [ garbage ]` also returns `[ garbage ]`.